### PR TITLE
test quota sync with LDAP

### PR DESCRIPTION
**Pull request by individual-it**
ToDo: 
- [x]  stash commits before merging

### CODE REVIEW

File: tests/acceptance/features/webUIProvisioning/users.feature [hunk 1]
```diff
@@ -68,17 +69,85 @@ Feature: add users
...
+  @issue-core-33186
+  Scenario: admin sets quota of user when the default quota setting is set and quota LDAP attribute is specified
+    #to set Quota we can just misuse any LDAP text field
+    Given LDAP config "LDAPTestId" has key "ldapQuotaAttribute" set to "employeeNumber"
+    And LDAP config "LDAPTestId" has key "ldapQuotaDefault" set to "10MB"
+    When the administrator sets the ldap attribute "employeeNumber" of the entry "uid=user0,ou=TestUsers" to "11 MB"
+    And the LDAP users are resynced
+    And the administrator sets the quota of user "user0" to "13 MB" using the webUI
+    Then the quota definition of user "user0" should be "13 MB"
+    And the quota of user "user0" should be set to "13 MB" on the webUI
```
Comment (phil-davis) [inline tests/acceptance/features/webUIProvisioning/users.feature]: I think this needs commented-out examples of what the steps should be when the issue is fixed.

File: tests/acceptance/features/apiProvisioningLDAP/users.feature [hunk 2]
```diff
@@ -127,6 +127,97 @@ Feature: Manage users using the Provisioning API
...
+    #to set Quota we can just misuse any LDAP text field
+    And LDAP config "LDAPTestId" has key "ldapQuotaAttribute" set to "employeeNumber"
+    And LDAP config "LDAPTestId" has key "ldapQuotaDefault" set to "10MB"
+    And the LDAP users have been resynced
+    When the administrator changes the quota of user "user1" to "13MB" using the provisioning API
+    Then the OCS status code should be "<ocs-status-code>"
+    And the HTTP status code should be "<http-status-code>"
+    And the quota definition of user "user1" should be "13 MB"
+    And the LDAP users are resynced
+    And the quota definition of user "user1" should be "10MB"
```
[the remark this answers is not in the document]
Comment (dpakach) [reply tests/acceptance/features/apiProvisioningLDAP/users.feature]: I think it also comes under #33186 so maybe we can tag this scenario also.

**Comment by individual-it** [comment]
@phil-davis @dpakach I've tagged the  scenarios and added more steps for when the bug is fixed, please review again.
I've made a new commit for the ease of reviewing. Please do not merge before I stashed the commits

**Comment by individual-it** [comment]
@tomneedham @pmaier1 please review these scenarios
I don't want to merge them before you confirm that that is what you expect and wrote in the changelog

File: tests/acceptance/features/apiProvisioningLDAP/users.feature [hunk 3]
```diff
@@ -129,6 +129,101 @@ Feature: Manage users using the Provisioning API
...
+      | ocs-api-version | ocs-status-code | http-status-code |
+      | 1               | 100             | 200              |
+      | 2               | 200             | 200              |
+
+  @issue-core-33186
+  Scenario Outline: admin sets quota of user for which no LDAP quota attribute is specified but the default quota setting is set
+    Given using OCS API version "<ocs-api-version>"
+    #to set Quota we can just misuse any LDAP text field
+    And LDAP config "LDAPTestId" has key "ldapQuotaAttribute" set to "employeeNumber"
+    And LDAP config "LDAPTestId" has key "ldapQuotaDefault" set to "10MB"
```
Comment (pmaier1) [inline tests/acceptance/features/apiProvisioningLDAP/users.feature]: In the scenario outline you say "for which **no** LDAP quota attribute is specified" but then you specify a LDAP quota attribute in this line? Do I misunderstand?

Comment (pmaier1) [reply tests/acceptance/features/apiProvisioningLDAP/users.feature]: Ah, it refers to the **default** quota attribute in LDAP, right?

File: tests/acceptance/features/webUIProvisioning/users.feature [hunk 4]
```diff
@@ -68,17 +69,85 @@ Feature: add users
...
+    Given LDAP config "LDAPTestId" has key "ldapQuotaAttribute" set to "employeeNumber"
+    And LDAP config "LDAPTestId" has key "ldapQuotaDefault" set to "10MB"
+    And the LDAP users have been resynced
+    When the administrator sets the quota of user "user0" to "13 MB" using the webUI
+    Then the quota definition of user "user0" should be "13 MB"
+    And the quota of user "user0" should be set to "13 MB" on the webUI
+    When the LDAP users are resynced
+    And the administrator reloads the users page
+    Then the quota definition of user "user0" should be "10MB"
+    And the quota of user "user0" should be set to "10MB" on the webUI
```
[the remark this answers is not in the document]
Comment (pmaier1) [reply tests/acceptance/features/webUIProvisioning/users.feature]: Technical limitation of core which we accept for now.

File: tests/acceptance/features/apiProvisioningLDAP/users.feature (hunk 3 above)
Comment (individual-it) [reply tests/acceptance/features/apiProvisioningLDAP/users.feature]: yes

**Comment by individual-it** [comment]
@pmaier1 thank you for your feedback. Its really useful to have the feedback from s.o. who does not work with that things every day. The feature files should be understandable for anyone.
1. I've adjusted the Scenario heading, hopefully now its clearer that in that particular cases its the default Quota set by the LDAP setting.
2. we have API tests and webUI tests. In this case we want to make sure that the API behaves correctly but also the webUI, so some tests are in both environments

**Pull request merged**